sections/header.liquid

```diff
@@ -490,6 +492,30 @@
       {%- endif -%}
     {%- endif -%}
 
+    {%- if section.settings.logo_position == 'middle-center' -%}
+      {%- if request.page_type == 'index' -%}
+        <h1 class="header__heading">
+      {%- endif -%}
+          <a href="{{ routes.root_url }}" class="header__heading-link link link--text focus-inset">
+            {%- if section.settings.logo != blank -%}
+              {%- assign logo_alt = section.settings.logo.alt | default: shop.name | escape -%}
```

Comment:
Not sure if this is what you were referring to, but yes, we do need to assign the variable first because it's being modified (`escape` and `default` filters)

If it was `section.settings.logo.alt`, we could simply pass it as that `alt: section.settings.logo.alt `, and passing it like this: `alt: section.settings.logo.alt | default: shop.name | escape` is invalid.